### PR TITLE
Implement `sphere_to_plane`

### LINKED ISSUE #181: Function to project sphere -> tangent plane

### PR DESCRIPTION
**Pull request by milancurcic**
* [x] Implementation
* [x] Tests
* [x] Docs

Closes #181.

### CODE REVIEW

**Comment by milancurcic** [comment]
On second thought, I'm unsure if I got this right. Should `x_origin` and `y_origin` be `lon_origin` and `lat_origin` instead? `x_origin` and `y_origin` don't seem useful.

File: clouddrift/sphere.py [hunk 1]
```diff
@@ -125,9 +125,24 @@ def sphere_to_plane(
...
+
+    Raises
+    ------
+    AttributeError
+        If ``lon`` and ``lat`` are not NumPy arrays
     """
-    x = np.empty_like(lon)
-    y = np.empty_like(lat)
+    x = np.zeros(lon.shape, dtype=lon.dtype)
+    y = np.zeros(lat.shape, dtype=lat.dtype)
```
Comment (philippemiron) [inline clouddrift/sphere.py:145]: x = np.zeros_like(lon)
y = np.zeros_like(lat)

File: clouddrift/sphere.py [hunk 2]
```diff
@@ -125,9 +125,24 @@ def sphere_to_plane(
...
+    (array([5.00000000e+01, 1.11318845e+05]), array([150.,   0.]))
+
+    Raises
+    ------
+    AttributeError
+        If ``lon`` and ``lat`` are not NumPy arrays
     """
-    x = np.empty_like(lon)
-    y = np.empty_like(lat)
+    x = np.zeros(lon.shape, dtype=lon.dtype)
```
Comment (philippemiron) [inline clouddrift/sphere.py:144]:
```suggestion
    x = np.zeros_like(lon)
```

File: clouddrift/sphere.py (hunk 1 above)
Comment (philippemiron) [inline clouddrift/sphere.py:145]:
```suggestion
    y = np.zeros(lat.shape, dtype=lat.dtype)
```
```suggestion
    x = np.zeros_like(lon)
    y = np.zeros_like(lat)
```

Comment (philippemiron) [inline clouddrift/sphere.py:145]:
```suggestion
    x = np.zeros_like(lon)
    y = np.zeros_like(lat)
```

**Comment by philippemiron** [comment]
> On second thought, I'm unsure if I got this right. Should `x_origin` and `y_origin` be `lon_origin` and `lat_origin` instead? `x_origin` and `y_origin` don't seem useful.

Yes. My understanding is that you set the origin, for example, `(0˚, 50˚N)` of a domain `[-10˚W, 10˚E]` and `[40˚N, 75˚N]` that will map to `[0,0]` in the tangent plane.

**Pull request merged**